File: DynamicProgramming1/RGBStreet/JavaRGBStreet/src/RGBStreet.java
import java.io.*;
import java.util.StringTokenizer;

public class RGBStreet {
    static final int MAX = 1000;

    static int[][] cost = new int[MAX + 1][3];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        for (int i = 1; i <= n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());

            cost[i][0] = Integer.parseInt(st.nextToken()) + Math.min(cost[i - 1][1], cost[i - 1][2]);
            cost[i][1] = Integer.parseInt(st.nextToken()) + Math.min(cost[i - 1][0], cost[i - 1][2]);
            cost[i][2] = Integer.parseInt(st.nextToken()) + Math.min(cost[i - 1][0], cost[i - 1][1]);
        }

        int minCost = Math.min(Math.min(cost[n][0], cost[n][1]), cost[n][2]);

        bw.write(minCost + "\n");

        br.close();
        bw.close();
    }
}

File: DynamicProgramming1/RGBStreet/CPPRGBStreet/RGBStreet.cpp
#include <iostream>
#include <algorithm>

#define MAX 1000

using namespace std;

struct Cost
{
	int R;
	int G;
	int B;
} cost[MAX + 1];

int main()
{
	int n;
	Cost temp;
	int min_cost;

	cin >> n;

	for (int i = 1; i <= n; i++)
	{
		cin >> temp.R >> temp.G >> temp.B;

		cost[i].R = temp.R + min(cost[i - 1].G, cost[i - 1].B);
		cost[i].G = temp.G + min(cost[i - 1].R, cost[i - 1].B);
		cost[i].B = temp.B + min(cost[i - 1].R, cost[i - 1].G);
	}

	min_cost = min(min(cost[n].R, cost[n].G), cost[n].B);

	cout << min_cost << "\n";

	return 0;
}
